src/Components/HomePage.jsx

```diff
@@ -30,15 +38,16 @@ export default function HomePage() {
                 setPageIsLoading(false)
                 setArticlesLoading(false)
             })
-    }, [topic_query])
+    }, [topic_query, searchFiltersApplied])
 
     if(pageIsLoading){
         return <p>Loading...</p>
     }
 
     return (
         <>
-        <TopicSelector setTopic={setTopic} topics={topics}/>
+        <TopicSelector setTopic={setTopic} topics={topics} setSearchParams={setSearchParams}/>
+        <SortFilter setSearchFiltersApplied={setSearchFiltersApplied} setSearchParams={setSearchParams} sortQuery={sortQuery} orderQuery={orderQuery} topic_query={topic_query}/>
         {topic_query === undefined ? <h2>Latest Articles</h2> : <h2>{topic_query}</h2>}
         {articlesLoading ? <p>Articles loading...</p> : 
         <ArticleCards articles={articles} topic={topic}/>}
```

Comment:
Sort by looks like it's working well, though it's hard to tell for sure when your article cards don't have things you can sort by on them, only the date is visible. Could you add the comment count and number of votes to the article cards to make this clearer for the user.